torch_geometric/data/batch.py

```diff
@@ -183,3 +184,65 @@ def num_graphs(self) -> int:
     def __reduce__(self):
         state = self.__dict__.copy()
         return DynamicInheritanceGetter(), self.__class__.__bases__, state
+
+    def subbatch(self, idx):
+        # convert index to mask
+        if isinstance(idx, slice) or \
+           (isinstance(idx, Tensor) and idx.dtype == torch.long) or \
```

Comment:
I think in most other parts of the [code](https://github.com/pyg-team/pytorch_geometric/blob/210ec4bde3dfdb9d587c0caddf2cef92bc557430/torch_geometric/data/batch.py#L120) I see that these kinds of conditions are usually split (even when matching result). Maybe we do the same here?